src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -185,6 +186,58 @@ public function orWhereDoesntHave($relation, Closure $callback = null)
         return $this->doesntHave($relation, 'or', $callback);
     }
 
+    /**
+     * Add a strict relationship condition to the query by its primary key.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model|string|array  $relation
+     * @param  mixed|null $id
```

Comment:
Mixed already contains `null`.

```suggestion
     * @param  mixed  $id
```